zipline/algorithm.py

```diff
@@ -2241,19 +2227,25 @@ def _run_pipeline(self, pipeline, start_date, chunksize):
         --------
         PipelineEngine.run_pipeline
         """
-        days = self.trading_schedule.all_execution_days
+        sessions = self.trading_calendar.all_sessions
 
         # Load data starting from the previous trading day...
-        start_date_loc = days.get_loc(start_date)
+        start_date_loc = sessions.get_loc(start_session)
 
         # ...continuing until either the day before the simulation end, or
         # until chunksize days of data have been loaded.
-        sim_end = self.sim_params.last_close.normalize()
-        end_loc = min(start_date_loc + chunksize, days.get_loc(sim_end))
-        end_date = days[end_loc]
+        sim_end_session = self.sim_params.period_end
```

Comment:
If this PR is about clarity of names, should we rename `period_end` to `session_end` to match the new calendar names?